packages/mephisto-worker-addons/src/Feedback/index.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect, useReducer } from "react";
+import { usePopperTooltip } from "react-popper-tooltip";
+import "./index.css";
+import { feedbackReducer } from "../Reducers";
+import FeedbackTextArea from "./FeedbackTextArea";
+import Question from "./Question";
+import SubmitButton from "./SubmitButton";
+
+function Feedback({
+  headless,
+  questions,
+  handleSubmit,
+  maxTextLength,
+  textAreaWidth,
+}) {
+  // To make classNames more readable
+  const headlessPrefix = headless ? "headless-" : "";
+  const stylePrefix = `${headlessPrefix}mephisto-worker-addons-feedback__`;
+  const stylePrefixNoHeadlessPrefix = `mephisto-worker-addons-feedback__`;
+
+  // Setting defaults
+  const maxFeedbackLength = maxTextLength ? maxTextLength : 700;
+  const modifiedTextAreaWidth = textAreaWidth ? textAreaWidth : "100%";
+  // For when there are questions
+  const [questionsFeedbackText, setQuestionsFeedbackText] = useState([]);
+
+  // For use when there are no questions
+  const [generalFeedbackText, setGeneralFeedbackText] = useState("");
+  const [state, dispatch] = useReducer(feedbackReducer, {
+    status: 0,
+    text: "",
+    errorIndexes: null,
+  });
```

Comment:
Handles different states that the feedback component can be in (error, loading, normal, etc...)